### ISSUE TITLE
Upgrading from v0.2.0-beta-3 to v0.2.0-beta-4 failed

### ISSUE THREAD

**Issue by winksaville** [paradigmxyz/reth#7366]
### Describe the bug

I was seeing this in the reth logs:
```
2024-03-27T18:02:49.203999Z  INFO reth::cli: Status connected_peers=91 freelist=477846 latest_block=1221399
2024-03-27T18:03:10.721586Z  WARN reth_node_core::events::node: Beacon client online, but no consensus updates received for a while. Please fix your beacon client to follow the chain! period=88129.630204334s
2024-03-27T18:03:14.204252Z  INFO reth::cli: Status connected_peers=90 freelist=477846 latest_block=1221399
2024-03-27T18:03:39.204621Z  INFO reth::cli: Status connected_peers=90 freelist=477846 latest_block=1221399
...
2024-03-27T18:07:49.205065Z  INFO reth::cli: Status connected_peers=90 freelist=477846 latest_block=1221399
2024-03-27T18:08:10.723179Z  WARN reth_node_core::events::node: Beacon client online, but no consensus updates received for a while. Please fix your beacon client to follow the chain! period=88429.631813739s
2024-03-27T18:08:14.204841Z  INFO reth::cli: Status connected_peers=90 freelist=477846 latest_block=1221399
2024-03-27T18:08:39.204334Z  INFO reth::cli: Status connected_peers=90 freelist=477846 latest_block=1221399
```
And this lighthouse bn logs:
```
Mar 27 18:02:48.001 WARN Error processing HTTP API request, method: POST, path: /eth/v1/validator/duties/attester/40781, status: 400 Bad Request, elapsed: 236.046µs, module: http_api:215
Mar 27 18:02:48.001 WARN Error processing HTTP API request, method: POST, path: /eth/v1/validator/prepare_beacon_proposer, status: 400 Bad Request, elapsed: 166.311µs, module: http_api:215
Mar 27 18:02:48.001 WARN Error processing HTTP API request, method: GET, path: /eth/v1/validator/duties/proposer/40781, status: 503 Service Unavailable, elapsed: 353.102µs, module: http_api:215
Mar 27 18:02:48.004 WARN Error processing HTTP API request, method: POST, path: /eth/v1/validator/duties/attester/40782, status: 400 Bad Request, elapsed: 167.852µs, module: http_api:215
Mar 27 18:02:54.001 INFO Syncing, est_time: --, distance: 7389 slots (1 day 0 hrs), peers: 54, service: slot_notifier, module: client::notifier:255
Mar 27 18:02:54.001 WARN Syncing deposit contract block cache, est_blocks_remaining: 5409, service: slot_notifier, module: client::notifier:656
Mar 27 18:02:56.608 WARN Failed to sync chain built on invalid parent, advice: check execution node for corruption then restart it and Lighthouse, parent_root: 0x2223b44c4814a2b6948d5e6ef48c80d9d0ea7851f2bf655fc23981be5133250b, module: network::network_beacon_processor::sync_methods:729
Mar 27 18:03:00.001 WARN Error processing HTTP API request, method: POST, path: /eth/v1/validator/duties/attester/40781, status: 400 Bad Request, elapsed: 294.588µs, module: http_api:215
Mar 27 18:03:00.001 WARN Error processing HTTP API request, method: GET, path: /eth/v1/validator/duties/proposer/40781, status: 503 Service Unavailable, elapsed: 422.85µs, module: http_api:215
Mar 27 18:03:00.002 WARN Error processing HTTP API request, method: POST, path: /eth/v1/validator/prepare_beacon_proposer, status: 400 Bad Request, elapsed: 393.823µs, module: http_api:215
Mar 27 18:03:00.005 WARN Error processing HTTP API request, method: POST, path: /eth/v1/validator/duties/attester/40782, status: 400 Bad Request, elapsed: 198.015µs, module: http_api:215
Mar 27 18:03:06.000 INFO Syncing, est_time: --, distance: 7390 slots (1 day 0 hrs), peers: 57, service: slot_notifier, module: client::notifier:255
Mar 27 18:03:06.001 WARN Syncing deposit contract block cache, est_blocks_remaining: 5409, service: slot_notifier, module: client::notifier:656
Mar 27 18:03:07.894 WARN Failed to sync chain built on invalid parent, advice: check execution node for corruption then restart it and Lighthouse, parent_root: 0x2223b44c4814a2b6948d5e6ef48c80d9d0ea7851f2bf655fc23981be5133250b, module: network::network_beacon_processor::sync_methods:729
Mar 27 18:03:12.001 WARN Error processing HTTP API request, method: POST, path: /eth/v1/validator/prepare_beacon_proposer, status: 400 Bad Request, elapsed: 175.887µs, module: http_api:215
```

I saw that there was v0.2.0-beta-4 out so I thought I'd upgrade my v0.2.0-beta-3. So I stopped reth, compiled beta-4 and restarted I reth which went into a shutdown/restart loop:
```
2024-03-27T18:07:49.205065Z  INFO reth::cli: Status connected_peers=90 freelist=477846 latest_block=1221399
2024-03-27T18:08:10.723179Z  WARN reth_node_core::events::node: Beacon client online, but no consensus updates received for a while. Please fix your beacon client to follow the chain! period=88429.631813739s
2024-03-27T18:08:14.204841Z  INFO reth::cli: Status connected_peers=90 freelist=477846 latest_block=1221399
2024-03-27T18:08:39.204334Z  INFO reth::cli: Status connected_peers=90 freelist=477846 latest_block=1221399
2024-03-27T18:09:04.204255Z  INFO reth::cli: Status connected_peers=90 freelist=477846 latest_block=1221399
2024-03-27T18:09:29.204043Z  INFO reth::cli: Status connected_peers=90 freelist=477846 latest_block=1221399
2024-03-27T18:09:54.204492Z  INFO reth::cli: Status connected_peers=91 freelist=477846 latest_block=1221399
2024-03-27T18:10:19.204361Z  INFO reth::cli: Status connected_peers=90 freelist=477846 latest_block=1221399
2024-03-27T18:10:31.022594Z  INFO reth::cli: Wrote network peers to file peers_file="/home/kendall/eth2-data/reth/holesky/known-peers.json"
2024-03-27T18:10:56.824702Z  INFO reth::cli: Starting reth version="0.2.0-beta.4 (c04dbe6e9)"
2024-03-27T18:10:56.825434Z  INFO reth::cli: Opening database path="/home/kendall/eth2-data/reth/holesky/db"
2024-03-27T18:10:56.860957Z  INFO reth::cli: Configuration loaded path="/home/kendall/eth2-data/reth/holesky/reth.toml"
2024-03-27T18:10:56.868829Z  INFO reth::cli: Database opened
2024-03-27T18:10:56.869594Z  INFO reth::cli: 
2024-03-27T18:10:56.973882Z  INFO reth::cli: Transaction pool initialized
2024-03-27T18:10:56.973927Z  INFO reth::cli: Connecting to P2P network
2024-03-27T18:10:56.976755Z  INFO net::peers: Loading saved peers file=/home/kendall/eth2-data/reth/holesky/known-peers.json
2024-03-27T18:10:56.996588Z  INFO reth::cli: StaticFileProducer initialized
2024-03-27T18:10:56.996805Z  INFO reth::cli: Pruner initialized prune_config=PruneConfig { block_interval: 5, segments: PruneModes { sender_recovery: Some(Full), transaction_lookup: None, receipts: Some(Before(0)), account_history: Some(Distance(10064)), storage_history: Some(Distance(10064)), receipts_log_filter: ReceiptsLogPruneConfig({0x4242424242424242424242424242424242424242: Before(0)}) } }
2024-03-27T18:10:56.996951Z  INFO reth::cli: Consensus engine initialized
2024-03-27T18:10:56.997006Z  INFO reth::cli: Engine API handler initialized
2024-03-27T18:10:56.999210Z  INFO reth::cli: RPC auth server started url=127.0.0.1:8551
2024-03-27T18:10:56.999469Z  INFO reth::cli: RPC IPC server started url=/tmp/reth.ipc
2024-03-27T18:10:56.999495Z  INFO reth::cli: RPC HTTP server started url=0.0.0.0:8545
2024-03-27T18:10:56.999500Z  INFO reth::cli: RPC WS server started url=127.0.0.1:8546
2024-03-27T18:10:56.999508Z  INFO reth::cli: Starting consensus engine
2024-03-27T18:10:59.032743Z  INFO reth_node_core::events::node: Live sync in progress, downloading blocks remaining_blocks=1 target_block_hash=0xc03268d3c36799e7b5314c7cc754d763b21430ad3463e35e989dff670a30158d

...

2024-03-27T18:11:24.520886Z  INFO reth_node_core::events::node: Preparing stage pipeline_stages=8/12 stage=MerkleExecute checkpoint=1221399 target=1221413
2024-03-27T18:11:24.520902Z  INFO reth_node_core::events::node: Executing stage pipeline_stages=8/12 stage=MerkleExecute checkpoint=1221399 target=1221413
2024-03-27T18:11:24.997634Z  INFO reth::cli: Status connected_peers=2 freelist=477566 stage=MerkleExecute checkpoint=1221399 target=1221413
2024-03-27T18:11:42.777995Z  WARN sync::stages::merkle: Failed to verify block state root target_block=1221413 got=0xbdc22791536473f93550a286d655f4cf2eba8dddf5e7978705f5b94cc1986f35 expected=SealedHeader { hash: 0x13ae715b236b106ec4f6b4940c3797c67f0bc2a915686a2b711bf1eeb83ff5b6, header: Header { parent_hash: 0x99df389bb0bf5fbd85f90c602d7c82b02ab35859319f4dd3fea40e4ea0c85b5f, ommers_hash: 0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347, beneficiary: 0xea7e42c7c9262e6831bfa7be71d67aa5523e27bd, state_root: 0x35817486e19aa208e8818095f6614a18d887d2f90a793a14b45dcf17838f4ab3, transactions_root: 0xd3288555e0d7452b4982a889b42e4bd43ea13163df0ec1e253c13032e79fd6d2, receipts_root: 0x2134a54444b048160a200f67269443804a2a40e05af40d32edbca306c99c53f3, withdrawals_root: Some(0x8a5e2e8ba60d6eebed4bd8917739b957792466fafb0b253d9fc965977721cf59), logs_bloom: 0xd7193857ab8354d69b6893124250c2834282b02752ab7f1d4fb084c1005502f014027149b1103a0a38443d2418617a8851446009a5514446530c7a351c78a1580ecc531d8414890446030a7c3cca280b613bb8422cc480b00c2e820812c202f268cdc870deb351c8f58168d55eac9c03507c060d04d1e1718a10163667f2180852ef2460b015b4d49381d321b0890790baaa3931127560f0750d163b0c86554677521da8a48c858400824088c5c86e31efd20c24cc4d0606d9aa342dc8c6219a0259b80f71321aa9f060982c0e9d45106e503713e2e338d4198c3ab8e608a52933388269570d15daaf6e03688a298073cd86d0dc21affd1f5c322a138c320b4a, difficulty: 0x0_U256, number: 1221413, gas_limit: 30000000, gas_used: 29014208, timestamp: 1711473216, mix_hash: 0x1502d0ef06a12077e3baba5c2ef383489ad77c154449117240cac069cae42c0a, nonce: 0, base_fee_per_gas: Some(14391), blob_gas_used: Some(655360), excess_blob_gas: Some(79429632), parent_beacon_block_root: Some(0x03ef12b9fec56a0942269d7ba2ece7b23d8ee6fcedf06f49be8989a42b176a84), extra_data: 0x } }
2024-03-27T18:11:42.778133Z ERROR sync::pipeline: Stage encountered a validation error: mismatched block state root: got 0xbdc22791536473f93550a286d655f4cf2eba8dddf5e7978705f5b94cc1986f35, expected 0x35817486e19aa208e8818095f6614a18d887d2f90a793a14b45dcf17838f4ab3 stage=MerkleExecute bad_block=1221413
2024-03-27T18:11:42.832216Z  INFO Unwinding{[3mstage[0m[2m=[0mStorageHashing}: sync::pipeline: Stage unwound stage=StorageHashing unwind_to=1221399 progress=1221399 done=true
2024-03-27T18:11:43.297220Z  INFO Unwinding{[3mstage[0m[2m=[0mAccountHashing}: sync::pipeline: Stage unwound stage=AccountHashing unwind_to=1221399 progress=1221399 done=true
2024-03-27T18:11:43.573876Z  WARN Unwinding{[3mstage[0m[2m=[0mMerkleUnwind}: sync::stages::merkle: Failed to verify block state root target_block=1221399 got=0x49b6356fb5b5dcdfa516d4dfc0bebce4784ff3f9ecb4fb80d0d33cad7615318b expected=SealedHeader { hash: 0xb211ed799d6b6fc2acfa9b71da8339127ec7fa397e3921868e0d906fcdbc4773, header: Header { parent_hash: 0x7036555d8507594781176f125dd8a19ca8e16ea56a082d168b71d6f5a4dad5d9, ommers_hash: 0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347, beneficiary: 0x65fdc9a172746d54a5b05e6ac5130b991abaf079, state_root: 0xbfed25bec73befb55f37acb38924d7ad38eb30bc1e3ec2168dc64e6b89a5aafd, transactions_root: 0x715b36591bb834b33050020e5557c555ccdc9ea951fc1465ddcdd1bdb411b27c, receipts_root: 0xefd2b8ba9cb5154d75e59bdb017226ac21b0cbb74539e3f6e07f35ef7dbd9e20, withdrawals_root: Some(0x39b4b7fe42a5d6387b8495d2408c60d050267fd7574c84d1e600d988e9a2b4ec), logs_bloom: 0x8f0a049368a18c309cf8a19746818e004e00e4886586d13dad18e0ce461445098022891cc04cc01e060e60363010c05652118463c8ed1e90c18d8224e0a01b038015988020a790c38418b8d9a2790810093082270a1c582c869da1292e8c10aa9214c0826b0e0824dc00001448019c419024006511060ca34768091488722d3323b80d727454982071704a0c80403802505fa3f000a4150706a0101e4a68ec7257388b670281c140a832668405d181df0b4248205a4609910006c216201e0824533d4982011869101a0166018e481540f835a463ad2a45a0241910ab0ea26184423cd11c52840d42681d1bea526d712bbc8088f8ac1171ea2518a3cbcc610170, difficulty: 0x0_U256, number: 1221399, gas_limit: 30000000, gas_used: 22247415, timestamp: 1711473036, mix_hash: 0x330780942d6e3c24853561c1e8c17755927f7d1d1eb930e4106e128e1d64b939, nonce: 0, base_fee_per_gas: Some(14448), blob_gas_used: Some(524288), excess_blob_gas: Some(79560704), parent_beacon_block_root: Some(0x3cfc8447fbe250b17d276e22a06172e2cc1a934edfb15b968b06909f2b0c3917), extra_data: 0xd883010d0b846765746888676f312e32312e36856c696e7578 } }
2024-03-27T18:11:43.575333Z ERROR reth::cli: shutting down due to error
2024-03-27T18:11:48.724269Z  INFO reth::cli: Starting reth version="0.2.0-beta.4 (c04dbe6e9)"
2024-03-27T18:11:48.724939Z  INFO reth::cli: Opening database path="/home/kendall/eth2-data/reth/holesky/db"
2024-03-27T18:11:48.750528Z  INFO reth::cli: Configuration loaded path="/home/kendall/eth2-data/reth/holesky/reth.toml"
2024-03-27T18:11:48.755984Z  INFO reth::cli: Database opened
2024-03-27T18:11:48.756553Z  INFO reth::cli: 
2024-03-27T18:11:48.846721Z  INFO reth::cli: Transaction pool initialized
2024-03-27T18:11:48.846755Z  INFO reth::cli: Connecting to P2P network
2024-03-27T18:11:48.846893Z  INFO net::peers: Loading saved peers file=/home/kendall/eth2-data/reth/holesky/known-peers.json
2024-03-27T18:11:48.848166Z  INFO reth::cli: StaticFileProducer initialized
2024-03-27T18:11:48.848355Z  INFO reth::cli: Pruner initialized prune_config=PruneConfig { block_interval: 5, segments: PruneModes { sender_recovery: Some(Full), transaction_lookup: None, receipts: Some(Before(0)), account_history: Some(Distance(10064)), storage_history: Some(Distance(10064)), receipts_log_filter: ReceiptsLogPruneConfig({0x4242424242424242424242424242424242424242: Before(0)}) } }
2024-03-27T18:11:48.848469Z  INFO reth::cli: Consensus engine initialized
2024-03-27T18:11:48.848503Z  INFO reth::cli: Engine API handler initialized
2024-03-27T18:11:48.850618Z  INFO reth::cli: RPC auth server started url=127.0.0.1:8551
2024-03-27T18:11:48.850718Z  INFO reth::cli: RPC IPC server started url=/tmp/reth.ipc
2024-03-27T18:11:48.850728Z  INFO reth::cli: RPC HTTP server started url=0.0.0.0:8545
2024-03-27T18:11:48.850736Z  INFO reth::cli: RPC WS server started url=127.0.0.1:8546
2024-03-27T18:11:48.850760Z  INFO reth::cli: Starting consensus engine
2024-03-27T18:11:48.851466Z  INFO reth_node_core::events::node: Static File Producer started targets=StaticFileTargets { headers: None, receipts: None, transactions: None }
```

### Steps to reproduce

ATM, this is 100% reproducible on my node just booting, but I believe "root" problem is something happened in the past corrupting something that now causes reth to be unable to properly start.

Here is my node hardware/software:
```
kendall@ethel 24-03-27T23:48:49.583Z:~
$ ~/bin/reth --version && lighthouse --version && inxi && lsb_release -a && df -h
reth Version: 0.2.0-beta.4
Commit SHA: c04dbe6e9
Build Timestamp: 2024-03-27T17:50:28.869727319Z
Build Features: jemalloc
Build Profile: release
Lighthouse v5.1.2-5ce1619
BLS library: blst-modern
SHA256 hardware acceleration: false
Allocator: jemalloc
Profile: maxperf
Specs: mainnet (true), minimal (false), gnosis (true)
CPU: quad core Intel Core i5-8259U (-MT MCP-) speed/min/max: 425/400/3800 MHz
Kernel: 6.5.0-26-generic x86_64 Up: 2h 8m Mem: 1341.2/31953.5 MiB (4.2%)
Storage: 2.73 TiB (7.4% used) Procs: 248 Shell: Bash inxi: 3.3.13
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.4 LTS
Release:	22.04
Codename:	jammy
Filesystem      Size  Used Avail Use% Mounted on
tmpfs           3.2G  1.9M  3.2G   1% /run
/dev/nvme0n1p2  916G   57G  813G   7% /
tmpfs            16G     0   16G   0% /dev/shm
tmpfs           5.0M  4.0K  5.0M   1% /run/lock
efivarfs        192K   59K  129K  32% /sys/firmware/efi/efivars
/dev/nvme0n1p1  511M  6.1M  505M   2% /boot/efi
/dev/sda1       1.8T  151G  1.6T   9% /home/kendall/eth2-data
tmpfs           3.2G  108K  3.2G   1% /run/user/125
tmpfs           3.2G   96K  3.2G   1% /run/user/1000
```

The logs are [here as a tar.gz file](https://drive.google.com/file/d/1i8ed8oMPpwZmuwVYYndT2RKjj5wRvDtA/view?usp=drive_link).

### Platform(s)

Linux (x86)

### What version/commit are you on?

```
kendall@ethel 24-03-27T23:47:38.918Z:~
$ ~/bin/reth --version
reth Version: 0.2.0-beta.4
Commit SHA: c04dbe6e9
Build Timestamp: 2024-03-27T17:50:28.869727319Z
Build Features: jemalloc
Build Profile: release
```

### What database version are you on?

```
kendall@ethel 24-03-27T23:47:41.488Z:~
$ ~/bin/reth db version
Current database version: 2
Local database is uninitialized
```

### What type of node are you running?

Full via --full flag

### What prune config do you use, if any?

```
kendall@ethel 24-03-27T23:47:43.795Z:~
$ cat ~/eth2-data/reth/holesky/reth.toml
[stages.headers]
downloader_max_concurrent_requests = 100
downloader_min_concurrent_requests = 5
downloader_max_buffered_responses = 100
downloader_request_limit = 1000
commit_threshold = 10000

[stages.bodies]
downloader_request_limit = 200
downloader_stream_batch_size = 1000
downloader_max_buffered_blocks_size_bytes = 2147483648
downloader_min_concurrent_requests = 5
downloader_max_concurrent_requests = 100

[stages.sender_recovery]
commit_threshold = 5000000

[stages.execution]
max_blocks = 500000
max_changes = 5000000
max_cumulative_gas = 1500000000000
max_duration = "10m"

[stages.account_hashing]
clean_threshold = 500000
commit_threshold = 100000

[stages.storage_hashing]
clean_threshold = 500000
commit_threshold = 100000

[stages.merkle]
clean_threshold = 50000

[stages.transaction_lookup]
chunk_size = 5000000

[stages.index_account_history]
commit_threshold = 100000

[stages.index_storage_history]
commit_threshold = 100000

[peers]
refill_slots_interval = "5s"
trusted_nodes = []
connect_trusted_nodes_only = false
max_backoff_count = 5
ban_duration = "12h"

[peers.connection_info]
max_outbound = 100
max_inbound = 30
max_concurrent_outbound_dials = 15

[peers.reputation_weights]
bad_message = -16384
bad_block = -16384
bad_transactions = -16384
already_seen_transactions = 0
timeout = -4096
bad_protocol = -2147483648
failed_to_connect = -25600
dropped = -4096
bad_announcement = -1024

[peers.backoff_durations]
low = "30s"
medium = "3m"
high = "15m"
max = "1h"

[sessions]
session_command_buffer = 32
session_event_buffer = 260

[sessions.limits]

[sessions.initial_internal_request_timeout]
secs = 20
nanos = 0

[sessions.protocol_breach_request_timeout]
secs = 120
nanos = 0
```

### If you've built Reth from source, provide the full command you used

cargo install --locked --path bin/reth --bin reth

### Code of Conduct

- [X] I agree to follow the Code of Conduct

**Comment by winksaville** [comment]
I'll leave my node stopped and in the current state at least until tomorrow morning, 0900 UTC-7, when I'll resync if I haven't heard anything more.

**Comment by gakonst** [comment]
Fixed in #7753.
